### PR TITLE
Rnaseq corr

### PR DESCRIPTION
**Pull request by ajlee21**
The main change in this PR is to add a notebook to perform correlation analysis of toy RNA-seq dataset (`2_correlation_genes.ipynb`) and add modules to support this analysis in `utils.py`

Some minor changes:
- Fix typo in the readme
- Plot validation of SRA data in PCA space in addition to UMAP to verfiy that PC1 corresponds to difference in strains as expected.

### CODE REVIEW

File: sra_experiment/2_correlation_genes.ipynb [hunk 1]
```diff
@@ -0,0 +1,1387 @@
...
...WaN4ecHNFyt6GtXr0af39/3Nzc6N+/P0eOHAHKryxnz56Nm5sbAwcO5PTp09p1nJ2duVKpnn3u3LlER0dz6dIlBgwYAICnpydjx44lMDCQ1NRUJk+ejJubGyqVqkoMmzdvJiQkBE9PT1566SWdR7j99ttvDBgwAHd3dxYsWIAYzvzRJxKpINRBaioUF9/5irSsrLybjNAwkpOTWb9+PZs3byY+Pp6vvvpK+9SSAwcOMHDgQE6cOEHfvn3597//fcfy2rZty48//gjA8ePHWbduHfv378fOzo6YmBji4+Mxuq3ufv/+/axatYpPP/2UI0eO4O7uzsyZMwHIyckhMjKSGTNm8Mcff9CmTRv+/PPPBt4LwsNGJFJBqIOKri+1XZG2aFH+V1TvNhy5XI5KpeLixYuo1WocHBxo898m0u7u7gQEBCCXy4mIiOD8+fP3JYYNGzYwceJE2rdvj0KhYPLkySQmJpKens4vv/xChw4dGDBgAIaGhrzwwgu0qPggCI8skUgFoQ4qur7UdkXavHn53+vX7388jwsnJyfmzZvHihUr8PX15bXXXiMrKwtAJ2GZmJigVCopLS1t8BgyMjJYvHgxHh4eeHh44OXlhSRJZGVlkZ2dTevWrbXLymQybG1tGzwG4eEiWu0KQh0kJYGFBdR2jmzWrHzoQJFIG1ZYWBhhYWEUFBTw1ltv8eGHH2qvSmtiampKcXGx9vW1a9doVXmA5Htga2vL5MmTCQ8PrzLvypUrXK00CockSWRmZtZpO0LjIa5IBaEO/v67/Gq08hi7t1MooGlTUbXbkJKTkzly5AgqlQojIyOMjY0xMLjzaaxTp078+OOPaDQafvnlF44fP17nGEaNGsXq1au5cOECAPn5+ezatQuAgIAALly4wN69eyktLWXdunVcF7+kHnniilQQ6iApCXr0uPNyLVo8Glek+uznWZlKpWLZsmVcvHgRQ0ND3NzcWLBgARs3bqx1vfnz5zN37lzWr19PYGAggYGBdY4hKCiIwsJCoqKiSE9Px9LSEl9fX0JCQmjWrBkff/wxixYt4o033iAiIoLu3bvXeVtC4yCTRNtsAIYOHUpsbKy+wxAageJiMDeHt98u/wewenX1y379dfnV65Il5a8nTnwwMdZHYmIiLi4u+g5DeMjU9LkQ505RtSsI9+ziRZCk2hsaVWjRAvLyQK2+/3
```
Comment (jjc2718) [inline sra_experiment/2_correlation_genes.ipynb:1192]: I think this should be something like "What are the accessory genes with high correlation?" (not sure how to do suggested changes on Jupyter notebooks)

Comment (jjc2718) [inline sra_experiment/2_correlation_genes.ipynb:1350]: This is neat! Interested to hear if you find a biological reason for the high correlations in PA14 samples/PA14 reference.

File: core_acc_modules/utils.py [hunk 2]
```diff
@@ -5,21 +5,153 @@
 Scripts to process expression data for pilot experiment
 """
 import pandas as pd
+import random
 import gzip
 from Bio import SeqIO
 
+random.seed(123)
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply core_acc_modules/utils.py:12]: Very good point. I did not realize this. I went a little seed-crazy to make sure everything was reproducible. This makes sense

File: core_acc_modules/utils.py [hunk 3]
```diff
@@ -5,21 +5,153 @@
...
+
+    reference_genotype: str
+        Either 'pao1' or 'pa14'
+
+    """
+
+    # Read gene annotation
+    gene_mapping = pd.read_csv(gene_annotation_file, sep=",", header=0)
+
+    # Accessory are genes that don't have a mapping to the PA14 ID
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply core_acc_modules/utils.py:36]: This is the same method for defining core and accessory genes as I used for the array data. The only difference is that for the array data, I only had one PAO1 reference whereas for the rnaseq data I have both a PAO1 reference and PA14 reference that I need to keep track of.

File: core_acc_modules/utils.py [hunk 4]
```diff
@@ -5,21 +5,153 @@
...
+            ~gene_mapping["PAO1_ID"].isin(acc_genes), "annotation"
+        ] = "core"
+
+        # Add column with number of genes mapped in case user
+        # would like to only consider genes with 1-1 mapping
+        gene_mapping["num_mapped_genes"] = (
+            gene_mapping["PA14_ID"].str.split(", ").str.len()
+        )
+
+        gene_mapping.set_index("PAO1_ID", inplace=True)
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply core_acc_modules/utils.py:52]: Really nice suggestion! Thank you

File: sra_experiment/2_correlation_genes.ipynb [hunk 5]
```diff
@@ -0,0 +1,1387 @@
...
+    "print(gene_expression_ref_pao1.shape)\n",
+    "print(gene_expression_ref_pa14.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get shuffled dataset\n",
+    "The real gene expression data is shuffled along the genes in order to disrupt any existing relationship between genes. This will serve as a negative control when examining the relationship between genes later."
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply sra_experiment/2_correlation_genes.ipynb:75]: Good point. future me would also appreciate this.

File: core_acc_modules/utils.py [hunk 6]
```diff
@@ -47,3 +190,40 @@ def dict_gene_num_to_ids(fasta_file):
...
+        05_PA14000-4-2_5-10-07_S2.CEL | 0.8533 | 0.7252 |...
+        54375-4-05.CEL                | 0.7789 | 0.7678 |...
+        ...                           | ...    | ...    |...
+
+    Returns
+    -------
+        Permuted expression data
+    """
+
+    # For each sample, shuffle gene values (i.e. shuffle values independently per sample) in order to
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply core_acc_modules/utils.py:215]: Good point. Will move this

File: sra_experiment/2_correlation_genes.ipynb (hunk 1 above)
Comment (ajlee21) [reply sra_experiment/2_correlation_genes.ipynb:1350]: me too!

**Pull request merged**